examples/classification/benchmark.py

```diff
@@ -0,0 +1,393 @@
+from examples.classification.helper import _BatchLoader, _load_random_data, load_data
+from sparseml.onnx.utils import (
+    get_tensor_dim_shape,
+    override_model_batch_size,
+    set_tensor_dim_shape,
+)
+from sparsezoo import Zoo
+
+
+DEEPSPARSE_ENGINE = "deepsparse"
+ORT_ENGINE = "onnxruntime"
+
+
+def _run_model(
+    args, model: Any, batch: Union[numpy.ndarray]
+) -> List[Union[numpy.ndarray]]:
+    # run model according to engine type
+    if args.engine == ORT_ENGINE:
+        outputs = model.run(
+            [out.name for out in model.get_outputs()],  # outputs
+            {model.get_inputs()[0].name: batch},  # inputs dict
+        )
+    else:  # deepsparse
+        outputs = model.run(batch)
+    return outputs
+
+
+def _get_data_loader(config, new_image_shape, total_iterations):
+    """
+    Helper function to get appropriate batch data loader
+    """
+    if not config.data_path:
+        data_loader = _load_random_data(
+            batch_size=config.batch_size,
+            iterations=total_iterations,
+            image_size=new_image_shape,
+        )
+    else:
+        dataset = load_data(config.data_path)
+        data_loader = _BatchLoader(dataset, config.batch_size, total_iterations)
+    return data_loader
+
+
+def fix_onnx_input_shape(
+    model_path: str,
+    image_shape: Optional[Tuple[int]],
+) -> Tuple[str, Optional[NamedTemporaryFile]]:
+    """
+    Creates a new ONNX model from the given path that accepts the given input
+    shape. If the given model already has the given input shape no modifications are
+    made. Uses a tempfile to store the modified model file.
+    :param model_path: file path to ONNX model or SparseZoo stub of the model
+        to be loaded
+    :param image_shape: 2-tuple of the image shape to resize this model to, or None if
+        no resizing needed
+    :return: filepath to an onnx model reshaped to the given input shape will be the
+        original path if the shape is the same.  Additionally returns the
+        NamedTemporaryFile for managing the scope of the object for file deletion.
+        Additionally returns the image-shape to benchmark the new model with.
+    """
+    original_model_path = model_path
+    if model_path.startswith("zoo:"):
+        # load SparseZoo Model from stub
+        model = Zoo.load_model_from_stub(model_path)
+        model_path = model.onnx_file.downloaded_path()
+        print(f"Downloaded {original_model_path} to {model_path}")
+
+    model = onnx.load(model_path)
+    model_input = model.graph.input[0]
+
+    original_x = get_tensor_dim_shape(model_input, 2)
+    original_y = get_tensor_dim_shape(model_input, 3)
+    original_image_shape = (original_x, original_y)
+
+    if not (isinstance(original_x, int) and isinstance(original_y, int)):
+        return (
+            model_path,
+            None,
+            original_image_shape,
+        )  # model graph does not have static integer input shape
```

Comment:
we don't need this check for this use case because we can override the dynamic shape with a static one. this was an issue for yolo because without a static shape, we couldn't correctly calculate the strides of the model to set a new corresponding output shape

Reply:
👍